### PR TITLE
UnixPB: Only Install CMake on mac where not already present.

### LINKED ISSUE #3549: GitHub Actions For Checking Ansible Playbooks
THe github action for checking the ansible playbooks on MacOS both have issues.

MacOS14 is failing due to the playbook attempting to install cmake on the github runner when it is already installed.

MacOS14 is also failing due to the keyserver.ubuntu.com hostname not resolving with a no route to host error ( raised upstream : actions/runner-images#9777 )

MacOS11 is performing so badly, that eventually the check appears to timeout, given that macos 11 was end of life back in November 2023, this check should be moved up to macos13.

### PR DESCRIPTION
**Pull request by steelhead31**
This should fix the failing GHA, as the MacOS 14 runner, already has cmake installed.

Fixes #3549 

##### Checklist

- [X] commit message has one of the [standard prefixes](https://github.com/adoptium/infrastructure/blob/master/CONTRIBUTING.md#commit-messages)
- [X] VPC/QPC not applicable for this PR

### CODE REVIEW

**Review by gdams — APPROVED** [review]

**Review by sxa — APPROVED** [review]
I think the location is different for x64 and aarch64, although given that we're only seeing the problem on one of them (AFAIK) I'm ok with this for now.

I'm curious though - do we know why this causes the problem? I'd have hoped that `state: present` would make it do the right thing and only install if it's not there - almost feels like cmake has been installed on the runners in a manner which isn't compatible with brew.
We should also consider that if it's an "unofficial" install we've no idea where it came from or what version it is.

**Comment by steelhead31** [comment]
> I think the location is different for x64 and aarch64, although given that we're only seeing the problem on one of them (AFAIK) I'm ok with this for now.
> 
> I'm curious though - do we know why this causes the problem? I'd have hoped that `state: present` would make it do the right thing and only install if it's not there - almost feels like cmake has been installed on the runners in a manner which isn't compatible with brew. We should also consider that if it's an "unofficial" install we've no idea where it came from or what version it is.

@sxa Im not sure how the github runners are constructed, Im assuming without ansible, but I can ask the question upstream, on my bug... an alternate solution for the github action is to remove it from the runner, so the playbook can install the correct version, as this only occurs on the github runner, it works as expected everywhere else.

**Comment by steelhead31** [comment]
Going to quickly test the removal from the runner, and allowing the playbook to run normally :)

**Comment by steelhead31** [comment]
Removing cmake from the runner is not straightforward, so the ansible check is currently a tidier solution, I'll ask some questions upstream.

File: .github/workflows/build_mac.yml [hunk 1]
```diff
@@ -27,6 +27,10 @@ jobs:
     - name: Install dependencies
       run: brew install ansible
 
+    - name: Replace keyserver address with IP address
+      run: |
+        sed -i '' 's/keyserver\.ubuntu\.com/185.125.188.27/g' ansible/playbooks/Supporting_Scripts/package_signature_verification.sh
```
Comment (karianna) [inline .github/workflows/build_mac.yml:32]: Why?

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/MacOSX.yml [hunk 2]
```diff
@@ -125,6 +125,21 @@
   with_items: "{{ Build_Tool_Packages }}"
   tags: build_tools
 
+# Install CMAKE Where It Doesnt exist
```
Comment (karianna) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/MacOSX.yml:128]:
```suggestion
# Install CMAKE Where It Doesn't exist
```

**Review by karianna — CHANGES_REQUESTED** [review]

**Comment by karianna** [comment]
Ran the Mac OS X 11 action twice but I can't see the clear reason for failure.

File: .github/workflows/build_mac.yml (hunk 1 above)
Comment (steelhead31) [reply .github/workflows/build_mac.yml:32]: The github runner, cant resolve keyserver.ubuntu.com, but the IP address is reachable:  upstream bug raised: actions/runner-images#9777

Comment (karianna) [reply .github/workflows/build_mac.yml:32]: Can we add a comment in the source code to that effect?

Comment (steelhead31) [reply .github/workflows/build_mac.yml:35]: Done!

Comment (steelhead31) [reply .github/workflows/build_mac.yml:35]: Done!

**Comment by steelhead31** [comment]
Also a quick note that the macos-11 runner will be deprecated from 28th June see https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners

```
The macos-11 label has been deprecated and will no longer be available after 6/28/2024.
```

**Comment by gdams** [comment]
> Also a quick note that the macos-11 runner will be deprecated from 28th June see https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners
> 
> ```
> The macos-11 label has been deprecated and will no longer be available after 6/28/2024.
> ```

I'd just kill it, maybe switch it out with macos-13?

**Comment by sxa** [comment]
> > Also a quick note that the macos-11 runner will be deprecated from 28th June see https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners
> > ```
> > The macos-11 label has been deprecated and will no longer be available after 6/28/2024.
> > ```
> 
> I'd just kill it, maybe switch it out with macos-13?

Yeah if it's being deprecated that seems like a reasonable course of action

**Pull request merged**